deepchem/molnet/load_function/load_dataset_template.py

```diff
@@ -0,0 +1,231 @@
+"""
+Short docstring description of dataset.
+"""
+import os
+import logging
+import deepchem
+from deepchem.feat import Featurizer
+from deepchem.trans import Transformer
+from deepchem.split.splitters import Splitter
+from deepchem.molnet.defaults import get_defaults
+
+from typing import List, Tuple, Optional
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_DIR = deepchem.utils.get_data_dir()
+MYDATASET_URL = 'http://deepchem.io.s3-website-us-west-1.amazonaws.com/datasets/mydataset.tar.gz'
+MYDATASET_CSV_URL = 'http://deepchem.io.s3-website-us-west-1.amazonaws.com/datasets/mydataset.csv'
+
+# Get dictionary of all featurizers, transformers, and splitters
+# Check this dict or `defaults.json` to see list of available classes
+DEFAULTS = get_defaults()
+
+# dict of accepted featurizers for this dataset
+# update for your dataset
+DEFAULT_FEATURIZERS = {
+    'AdjacencyFingerprint': deepchem.feat.AdjacencyFingerprint(),
+    'AtomicCoordinates': deepchem.feat.AtomicCoordinates(),
+    'ConvMolFeaturizer': deepchem.feat.ConvMolFeaturizer(),
+    'CoulombMatrix': deepchem.feat.CoulombMatrix(max_atoms=5),
+    'RDKitDescriptors': deepchem.feat.RDKitDescriptors(),
+    'RawFeaturizer': deepchem.feat.RawFeaturizer(),
+    'CircularFingerprint': deepchem.feat.CircularFingerprint(size=1024),
+}
+
+# dict of accepted transformers
+DEFAULT_TRANSFORMERS = {
+    'Power': deepchem.trans.PowerTransformer(),
+    'Balancing': deepchem.trans.BalancingTransformer(),
+    'Log': deepchem.trans.LogTransformer(),
+    'MinMax': deepchem.trans.MinMaxTransformer()
+}
+
+# dict of accepted splitters
+DEFAULT_SPLITTERS = {
+    'Index': deepchem.splits.IndexSplitter(),
+    'Random': deepchem.splits.RandomSplitter(),
+    'Scaffold': deepchem.splits.ScaffoldSplitter(),
+}
+
+
+def load_mydataset(
+    featurizer: Featurizer = DEFAULT_FEATURIZERS['RawFeaturizer'],
+    transformers: Tuple[Transformer] = (DEFAULT_TRANSFORMERS['Power']),
+    splitter: Splitter = DEFAULT_SPLITTERS['Random'],
+    reload: bool = True,
+    data_dir: Optional[str] = None,
+    save_dir: Optional[str] = None,
```

Comment:
Can we add 3 keyword arguments
```
featurizer_kwargs:Dict[str, object]
splitter_kwargs:Dict[str, object]
# This has a more complicated type since multiple transformers
transformer_kwargs:Dict[str, Dict[str, object]]
```
Users could use these to specify the parameters of their featurizers

Reply:
For example, `featurizer_kwargs={"size": 1024}` or `splitter_kwargs={"seed": 123}`